### PR TITLE
notification / menu fixes (Closes #8)

### LINKED ISSUE #8: A few plugin issues
- [x] onload connect / disconnect
- [x] notification icon CSS issue
- [x] page menu not showing up

https://github.com/samepage-network/obsidian-samepage/assets/3792666/6bb8829b-ed2a-49dd-9a97-3284ad2b3243

### PR DESCRIPTION
**Pull request by mdroidian**

### CODE REVIEW

File: .env.default [hunk 1]
```diff
@@ -0,0 +1 @@
+# OBSIDIAN_PLUGIN_PATH="C:\Users\YourUserName\ObsidianDirectory\.obsidian\plugins\samepage-dev"
```
Comment (dvargasfuertes) [inline .env.default:1]: Nice! -1 for windows convention :p

**Review by dvargasfuertes — COMMENTED** [review]
So the test failure isn't related to this PR, but wanna try taking a stab at resolving it? Time box yourself (~30 min) just to get familiar with testing and if not I could resolve once this is merged

**Comment by mdroidian** [comment]
> So the test failure isn't related to this PR, but wanna try taking a stab at resolving it? Time box yourself (~30 min) just to get familiar with testing and if not I could resolve once this is merged

So I was looking at the `"tabbing with spaces"` test, `runTest` then looking through `atJsonToObsidian.ts` and `leafParser.ts`
I was thinking of trying to find where in `obsidian-samepage` this conversion is being made.
I tried to run `npm test` but I got this:

```
error from playwright: Error: spawn npx ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn npx',
  path: 'npx',
  spawnargs: [
    'c8',
    '--reporter=lcov',
    '--reporter=text',
    '--all',
    '--include',
    'src',
    '--exclude',
    'node_modules',
    '--exclude-after-remap',
    'playwright',
    'test',
    '--config=./node_modules/samepage/testing/playwright.config.js'
  ]
}
Failed to run tests: Error: spawn npx ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn npx',
  path: 'npx',
  spawnargs: [
    'c8',
    '--reporter=lcov',
    '--reporter=text',
    '--all',
    '--include',
    'src',
    '--exclude',
    'node_modules',
    '--exclude-after-remap',
    'playwright',
    'test',
    '--config=./node_modules/samepage/testing/playwright.config.js'
  ]
}
```

but `spawn` doesn't seem to appear in the obsidian plugin as it did in samepage.network, so I wasn't sure how to fix it.

**Comment by dvargasfuertes** [comment]
damn, this might be another windows thing where we'll need to add the shell arg here: https://github.com/samepage-network/samepage.network/blob/main/package/scripts/test.ts#L30-L52

**Comment by mdroidian** [comment]
It looks like spacing is `undefined` here.

https://github.com/samepage-network/obsidian-samepage/blob/7e727c5cb1c275d871134292ac9682524dba0475/src/utils/atJsonToObsidian.ts#L59-L62

But I'm not quite sure why.

How can I find or see the value of `initialToken` ?
https://github.com/samepage-network/obsidian-samepage/blob/7e727c5cb1c275d871134292ac9682524dba0475/src/utils/leafParser.ts#L171-L173

**Comment by dvargasfuertes** [comment]
`spacing` being undefined wouldn't make sense to come from that post processor, since the return value from `replace()` will never be `undefined`.

What's more likely is that there's a `block` annotation that is not setting that `spacing` attribute at all that should be (when `level` > 1). Some clues are here:

https://github.com/samepage-network/obsidian-samepage/blob/7e727c5cb1c275d871134292ac9682524dba0475/src/utils/leafParser.ts#L74-L88

and here:

https://github.com/samepage-network/obsidian-samepage/blob/7e727c5cb1c275d871134292ac9682524dba0475/src/utils/leafParser.ts#L98-L111

**Comment by mdroidian** [comment]
Hmm.  Well changing

https://github.com/samepage-network/obsidian-samepage/blob/7e727c5cb1c275d871134292ac9682524dba0475/src/utils/atJsonToObsidian.ts#L61

to 

```
${level === 1 ? "" : spacing || "".concat("    ".repeat(level - 1)))}
```

seems to pass the `tabbing with spaces` test.  But it fails `Hello World Example` and `Single new line after line`

**Comment by mdroidian** [comment]
So I think I found the problem.  It look like the only way we handle `Indent using tabs` when we decode content back into Obsidian is via `appAttribute: {obdisian: { spacing`.  But Roam (and I'm guessing Logseq) don't add that attribute.
So the issue occurs in Obsidian, if you have setting `Editor` > `Indent using tabs` set to `off`.
- `encoding` from Obsidian adds successfully adds `appAttribute: {obdisian: { spacing`
- but when you add content in roam, no `appAttribute: {obdisian: { spacing`, defaults to tabs.
- resulting in:

image

Steps to repo
1. Share a blank page between obsidian and roam
2. Add bullet in obsidian
3. Add bullet in roam
4. Add bullet in obsidian

Is there an example of a plugin handling content related app settings I could look at?

**Comment by mdroidian** [comment]
I should have put the above message in a separate issue.  This PR is ready for review / merge.

**Review by dvargasfuertes — APPROVED** [review]

**Pull request merged**